src/db/migrations/init.sql

```diff
@@ -30,3 +31,11 @@ ON messages(channel_id);
 -- Create full-text search table.
 CREATE VIRTUAL TABLE messages_fts
 USING FTS5(content, username, avatar, messages_rowid);
+
+-- Create mapping between messages and channel pages.
+CREATE TABLE IF NOT EXISTS messages_pages (
```

Comment:
This table caches what page a given message belongs to in its channel. It costs ~2Mb but as a result retrieving the page surrounding a message takes ~4ms end-to-end (from http request to response). The next best query would cost ~160ms.

Reply:
Cool idea, probably wouldn't have thought of that 👍